src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int FIRST_LINE = 1;
+    private static final int OPERATION_CODE_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final String COMMA = ",";
+    private final List<FruitTransaction> fruitTransactions;
+    private String[] dataArray;
+    private FruitTransaction.Operation operation;
+
+    public DataConverterImpl() {
+        this.fruitTransactions = new ArrayList<>();
+    }
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> transactionData) {
+        if (transactionData.isEmpty()) {
+            throw new RuntimeException("CSV file is empty!" + transactionData);
```

Comment:
there is no File here
Delete this IF at all. For empty transactionData return empty List<FruitTransaction> without exception.

Reply:
I don't get, why? We can't work with empty data, then if it's empty it should be a misstake, no? The whole point of converting data is so we can read it and commit operations with it. What's the point of returning an empty list then?
Mb I'm mistaking, can you, please, explain this more to me?